unstable test TestExecConcurrent

```
cd util/async
go test -race -run TestExecConcurrent
--- FAIL: TestExecConcurrent (0.00s)
    runloop_test.go:152: 
                Error Trace:    /home/genius/project/src/github.com/tikv/client-go/util/async/runloop_test.go:152
                                                        /home/genius/project/go/src/runtime/asm_amd64.s:1700
                Error:          Received unexpected error:
                                runloop: already executing
                Test:           TestExecConcurrent
    runloop_test.go:158: 
                Error Trace:    /home/genius/project/src/github.com/tikv/client-go/util/async/runloop_test.go:158
                Error:          An error is expected but got nil.
                Test:           TestExecConcurrent
FAIL
exit status 1
FAIL    github.com/tikv/client-go/v2/util/async 0.013s
```

I'm not sure what this test is testing, the result is completely undeterminted.

--- a/util/async/runloop_test.go
+++ b/util/async/runloop_test.go
@@ -142,6 +142,9 @@ func TestExecCancelWhileWaiting(t *testing.T) {
 }
 
 func TestExecConcurrent(t *testing.T) {
+	defaultMaxProcs := runtime.GOMAXPROCS(0)
+	defer runtime.GOMAXPROCS(defaultMaxProcs)
+	runtime.GOMAXPROCS(1)
 	l := NewRunLoop()
 	l.Append(func() {
 		time.Sleep(time.Millisecond)

